test/porter/ingest_error_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+require 'aws-sdk-states'
+require 'json'
+
+step_functions = Aws::States::Client.new(
+  region: 'us-east-1'
+)
+
+describe :porter do
+  describe :ingesterror do
+    it 'returns the basic execution output' do
+      req = step_functions.start_execution({
+        state_machine_arn: CONFIG.PORTER_STATE_MACHINE_ARN,
+        input: {
+          Job: {
+            Id: 'porter-test-ingest-error',
+            Source: {
+              Mode: 'HTTP',
+              URL: 'http://example.com/404'
+            }
+          }
+        }.to_json
+      })
+
+      max_retries = 5
+      retries = 0
+
+      begin
+        desc = step_functions.describe_execution({
+          execution_arn: req.execution_arn,
+        })
+
+        raise RuntimeError if desc.status == 'RUNNING'
+      rescue RuntimeError => e
+        if retries <= max_retries
+          retries += 1
+          sleep 2 ** retries
```

Comment:
Although this test seems to consistently take more than 30 seconds ... why is that?

Reply:
Probably because in the dev env every lambda in the chain is coming out of a cold start